### PR TITLE
add interface for notifications, finish param checking implementation

### PR DESCRIPTION
**Pull request by mzanetti**

### CODE REVIEW

File: libguh/types/paramtype.h [hunk 1]
```diff
@@ -84,6 +87,15 @@ class LIBGUH_EXPORT ParamType
     bool m_readOnly;
 };
 
+class ParamTypes: public QList<ParamType>
+{
+public:
+    ParamTypes() = default;
+    ParamTypes(const QList<ParamType> &other);
+    ParamType findByName(const QString &name);
```
Comment (t-mon) [inline libguh/types/paramtype.h]: Note: the name changes if you change the language

File: libguh/types/paramtype.h [hunk 2]
```diff
@@ -84,6 +87,15 @@ class LIBGUH_EXPORT ParamType
...
+    ParamType findById(const ParamTypeId &id);
+};
+
 QDebug operator<<(QDebug dbg, const ParamType &paramType);
 QDebug operator<<(QDebug dbg, const QList<ParamType> &paramTypes);
```
Comment (t-mon) [inline libguh/types/paramtype.h]: Suggestion: QDebug operator<<(QDebug dbg, const ParamTypes &paramTypes);

File: libguh/types/paramtype.cpp [hunk 3]
```diff
@@ -213,3 +218,30 @@ QDebug operator<<(QDebug dbg, const QList<ParamType> &paramTypes)
...
 }
+
+ParamTypes::ParamTypes(const QList<ParamType> &other)
+{
+    foreach (const ParamType &pt, other) {
+        append(pt);
+    }
+}
+
+ParamType ParamTypes::findByName(const QString &name)
```
Comment (t-mon) [inline libguh/types/paramtype.cpp]: Note: the name changes if you change the language

File: libguh/types/paramtype.h (hunk 1 above)
Comment (mzanetti) [reply libguh/types/paramtype.h]: whaaaaaat?

Comment (mzanetti) [reply libguh/types/paramtype.h]: ok... turns out this is an issue indeed but as discussed offline we agreed to go with this for now, even if the interfaces stuff is broken when translations are active and we will get rid of translations in the json file in favor of interfaces when the apps actually start making use of interfaces.

File: libguh/types/paramtype.h (hunk 2 above)
Comment (mzanetti) [reply libguh/types/paramtype.h]: tots, will add

File: libguh/types/paramtype.cpp (hunk 3 above)
Comment (mzanetti) [reply libguh/types/paramtype.cpp]: same as above

File: libguh/types/paramtype.h (hunk 2 above)
Comment (mzanetti) [reply libguh/types/paramtype.h]: actually not needed as the implicit cast from ParamTypes to QList<ParamType> takes care of this.

**Pull request merged**